bin/pyspark

```
@@ -83,6 +83,8 @@ export PYTHONPATH="${SPARK_HOME}/python/lib/py4j-0.10.9.9-src.zip:$PYTHONPATH"
 export OLD_PYTHONSTARTUP="$PYTHONSTARTUP"
 export PYTHONSTARTUP="${SPARK_HOME}/python/pyspark/shell.py"
 
+export PYSPARK_DEFAULT_API_MODE=$($PYSPARK_DRIVER_PYTHON -c "from pyspark.util import default_api_mode; print(default_api_mode())")
```

if the module does not exist, this `export` will ignore the error and leave the variable unset?

---

and shall we reuse `SPARK_CONNECT_MODE`? https://github.com/apache/spark/pull/49930/files#diff-c174fde2f0bb75521b3507a31f0b600a65bb6855b20ed6e0bf42de5e9b55926eR154

---

I created this variable to indicate the spark connect distribution

---

Let me use some of the change from your PR as it's not merged yet.

---

> if the module does not exist, this export will ignore the error and leave the variable unset?

If the module does not exist, it will contain an error message from the Python command.
I'll add error check for the value, and show an error message and unset `SPARK_CONNECT_MODE`.

---

Now this covers the changes in #49930.